### ISSUE TITLE
Insights run now shows skips also which is a regression

### ISSUE THREAD

**Issue by Rohan21Lobo** [RedHatInsights/insights-core#3161]
After https://github.com/RedHatInsights/insights-core/pull/3154 , 

When insights run -p <rule_repo> is executed, it shows reponses of all types including SKIPS. So if the rules package has hundreds of rules, missing report will be shown from all of them 

Example 
~~~
# Before
(genv) ➜  ~ insights run -p myrules.check_errors ~/Downloads/sosrepor.tar.xz
---------
Progress:
---------
S

--------------
Rules Executed
--------------

----------------------
Rule Execution Summary
----------------------
Passed      : 0
Failed      : 0
Info        : 0
Missing Deps: 1
Fingerprint : 0
Metadata    : 0
Metadata Key: 0
Exceptions  : 0

# After
(genv) ➜  ~ cd ~/work/insights-core
(genv) ➜  insights-core git:(master) git cherry-pick 6dc1cd23b5427f5277b21ab0ff03c2a59f5a43ff
(genv) ➜  insights-core git:(master) insights run -p myrules.check_errors ~/Downloads/sosrepor.tar.xz
---------
Progress:
---------
S

--------------
Rules Executed
--------------
[SKIP] myrules.check_errors.report
------------------------------------------------------------------
Links:
    kcs:

Missing Dependencies:
    Requires:
        myrules.check_errors.is_rhel



----------------------
Rule Execution Summary
----------------------
Passed      : 0
Failed      : 0
Info        : 0
Missing Deps: 1
Fingerprint : 0
Metadata    : 0
Metadata Key: 0
Exceptions  : 0
~~~

**Comment by Rohan21Lobo** [comment]
@xiangce @bfahr

**Comment by bfahr** [comment]
@xiangce I reverted your PR.  Please take a look at the issue when you have time.

**Comment by xiangce** [comment]
@Rohan21Lobo - Thanks for reporting this bug.

@bfahr - I fixed it in this [commit-29de21f9](https://github.com/RedHatInsights/insights-core/pull/3175/commits/29de21f9ee799fc65f80f6d6e8f8f3c7f527a041) and submitted the patch with PR #3175, please have a review again. Thanks.
